packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -682,19 +683,45 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
     return linkMarks;
   }
 
+  isValidHref(url: string): boolean {
+    if (!this._autoLinkRegexNonGlobal?.test(url)) {
+      return false;
+    }
+
+    const href = this.buildHref(url);
+
+    if (!this.isValidTLD(href)) {
+      return false;
+    }
+
+    return true;
+  }
+
   private findAutoLinks(str: string): FoundAutoLink[] {
     const toAutoLink: FoundAutoLink[] = [];
 
-    for (const match of findMatches(str, this.options.autoLinkRegex)) {
-      const text = getMatchString(match);
+    const { autoLinkAfter, autoLinkRegex } = this.options;
+
+    let matcher = autoLinkRegex;
+
+    if (autoLinkAfter) {
+      matcher = new RegExp(autoLinkRegex.source + autoLinkAfter.source, 'gm');
```

Comment:
The original autolink regex has `gi` flags, not `gm`. Maybe it's important to have the flags match, or use the flags from the original?